src/workflow/snakemake/format/formatter.py

```diff
@@ -12,16 +17,43 @@ def format_output_templates_to_be_expanded(node):
     return outputs
 
 
-def extract_stages_from_path(path, known_stages):
+def format_input_templates_to_be_expanded(benchmark: Benchmark, wildcards):
+    """Formats benchmark inputs that will be expanded according to Snakemake's engine"""
+
+    pre = wildcards.pre
+    post = wildcards.post
+    name = wildcards.name
+
+    nodes = benchmark.get_nodes()
+    stage_ids = set([node.stage_id for node in nodes])
+
+    pre_stages = _extract_stages_from_path(pre, stage_ids)
+    after_stage = pre_stages[-1][0] if len(pre_stages) > 0 else None
+
+    stage_id, module_id, param_id = _match_node_format(post)
+
+    node_hash = hash(BenchmarkNode.to_id(stage_id, module_id, param_id, after_stage))
+    matching_node = next((node for node in nodes if hash(node) == node_hash), None)
+    assert matching_node is not None
+
+    node_inputs = matching_node.get_inputs()
+
+    inputs = _match_inputs(node_inputs, pre_stages, pre, name)
+
+    # print(f'Inputs: {stage_id} {module_id} {param_id}: {inputs}')
+    return inputs
+
+
+def _extract_stages_from_path(path: str, known_stage_ids: Set[str]):
     parts = path.split('/')
```

Comment:
```suggestion
    parts = [p for p in Path(path).parts]
```
In general, it is a good practice to use `from pathlib import Path` for path operations as it is system independent.